_docs/cardano/2017-01-04-transactions.md

```diff
@@ -3,41 +3,113 @@ layout: default
 title: Transactions in Cardano SL
 permalink: /cardano/transactions/
 group: cardano
-visible: false
 ---
 [//]: # (Reviewed at 403cea2d897aba95163b709bd13c35d343116f3f)
 
 # Transactions in Cardano SL
 
-You can think of transactions in Cardano SL as entities that consist of
-a list of inputs and a list of outputs. Outputs of a transaction can later
-be used as inputs for another transaction.
+## Overview
 
-Every node (except for SPV, or lightweight nodes) in the network
-verifies transactions, so those nodes have to keep track of unspent
-outputs, this is called “utxo”, or “Unspent Transaction Outputs”. By
-tracking utxo, every node can validate that inputs in a published
-transaction are indeed the unspent outputs.
+A transaction (_tx_) is a special data represents the _act_ of the value transferring 
+between two nodes (from the user's point of view - between two wallets). When the user 
+_Alice_ sends money to the user _Bob_, the new transaction emerges. Let's call this 
+transaction `Tx1`, the node under _Alice_' wallet `N1`, and the node under _Bob_'s wallet `N2`.
+
+So, the node `N1` do these steps:
+
+1. [creates](https://github.com/input-output-hk/cardano-sl/blob/63adb31e813e21ec9da21cfa69984840308bbfa2/src/Pos/Wallet/Tx.hs#L41) transaction `Tx1` and [signs](https://github.com/input-output-hk/cardano-sl/blob/63adb31e813e21ec9da21cfa69984840308bbfa2/src/Pos/Wallet/Tx/Pure.hs#L83) it with its private key,
+2. [sends](https://github.com/input-output-hk/cardano-sl/blob/63adb31e813e21ec9da21cfa69984840308bbfa2/src/Pos/Wallet/Tx.hs#L53) it to all known nodes (i.e. neighbors),
+3. [save](https://github.com/input-output-hk/cardano-sl/blob/63adb31e813e21ec9da21cfa69984840308bbfa2/src/Pos/Wallet/Tx.hs#L44) it in its local data.
+
+Each of `N1`'s neighbors sends transaction `Tx1` to its neighbors and so on, and some slot leader will store this transaction to the next block in the ledger.
+
+## Design
+
+Each transaction contains a list of _inputs_ and a list of _outputs_, and outputs of the transaction `Tx0` can be used as an inputs of the other transaction `Tx1` and so on:
+
+~~~
+            Tx0                           Tx1
+  +----------------------+      +----------------------+
+  |                      |      |                      |
+  |  Inputs     Outputs  |      |  Inputs     Outputs  |
+  | +------+   +-------+ |      | +------+   +-------+ |
+  | | In0  |   | Out0  + |      | | In0  |   | Out0  | |
+  | +------+   +-------+ |      | +------+   +-------+ |
+  | | In1  |   | Out1  | |      | | In1  |   | Out1  | |
+  | +------+   +-------+ |      | +------+   +-------+ |
+  | | ...  |   | ...   | |      | | ..   |   | ...   | |
+  | +------+   +-------+ |      | +------+   +-------+ |
+  | | InN  |   | OutN  | |      | | InN  |   | OutN  | |
```

Comment:
I think it's better to name last output as `OutM` to show that number of inputs and outputs can be different.